test/integration/controller/jobset_controller_test.go

```diff
@@ -516,6 +527,27 @@ func matchJobsSuspendState(js *jobset.JobSet, suspend bool) (bool, error) {
 	return true, nil
 }
 
+func matchJobsNodeSelectors(js *jobset.JobSet, nodeSelectors map[string]map[string]string) (bool, error) {
+	var jobList batchv1.JobList
+	if err := k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace)); err != nil {
+		return false, err
+	}
+	jobsChanged := 0
+	// Check if all jobs with name prefix js.Name+replicatedJobName have updated node selectors.
+	for _, job := range jobList.Items {
+		for index, nodeSelector := range nodeSelectors {
+			if strings.HasPrefix(job.Name, index) {
```

Comment:
we don't need two for loops, we can lookup the replciatedJobName from the labels and use it to lookup the expected node selectors


```
for _, job := range jobList.Items {
  wantSelectors := nodeSelectors[job.Labels[jobset.ReplicatedJobNameKey]]
 ...
}

```